src/core/iomgr/fd_posix.c

```diff
@@ -115,7 +121,10 @@ static void unref_by(grpc_fd *fd, int n) {
   gpr_atm old = gpr_atm_full_fetch_add(&fd->refst, -n);
   if (old == n) {
     close(fd->fd);
-    grpc_iomgr_add_callback(fd->on_done, fd->on_done_user_data);
+    fd->on_done_iocb->cb = fd->on_done;
```

Comment:
Mutate the interface to take a grpc_iomgr_closure and push the storage problem up one layer?